pkg/service/grpc_service.go

```diff
@@ -49,9 +52,16 @@ func (s *GRPCService) Serve(ctx context.Context, eval eval.IEvaluator) error {
 	grpcServer := grpc.NewServer(serverOpts...)
 	gen.RegisterServiceServer(grpcServer, s)
 
-	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", s.GRPCServiceConfiguration.Port))
-	if err != nil {
-		return err
+	if s.GRPCServiceConfiguration.ServerSocketPath != "" {
+		lis, err = net.Listen("unix", s.GRPCServiceConfiguration.ServerSocketPath)
+		if err != nil {
```

Comment:
These 2 error checks could be a single error check after the outer `if {} else {}` statement